### PR TITLE
Visualizer Shape Canvas

### LINKED ISSUE #257: Create UI for visualizer shapes
### Description of the task

Create a UI element that would hold all the shapes sent from the AI

### Outline the task in more detail

This UI element will display the shapes sent from the AI. It should take the form of an SVG element containing a number of svg shapes.

Shapes that need to be supported:
- Rect
- Circle
- Line
- Arc
- Poly

We should use the fields defined in #248 to figure out what needs to be rendered with each shape, but no logic should be attached with this task.

### Acceptance criteria

- [x] SVG component that holds our shapes
- [ ] SVG component should be in the main area of the UI
- [ ] Implementation for Rect, Circle, Line, Poly, and Arc
- [x] Example implementation for each shape
- [x] Ability to modify world unit in the SVG element
- [x] SVG element should adapt to the size of the window

### PR DESCRIPTION
**Pull request by qgolsteyn**
### Description

This PR adds a Canvas element, a Rect shape, and misc modifications, including:
- adds a storybook addon to improve visual testing (will come handy for testing arcs)
- update layer messages to match #248 and updates associated UI elements

### Testing Done

Manual testing

Closes #257 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/corner_kick/src/types/visualizer.ts [hunk 1]
```diff
@@ -1,9 +1,22 @@
-/**
+/*
  * This file describes a layer datatype
```
Comment (MathewMacDougall) [inline src/corner_kick/src/types/visualizer.ts]: Filename doesn't match file comment / purpose?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: src/corner_kick/src/types/visualizer.ts (hunk 1 above)
Comment (qgolsteyn) [reply src/corner_kick/src/types/visualizer.ts]: Fixed

File: src/corner_kick/src/components/Canvas/shapes/Rect.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    shape: IShape;
+}
+
+/**
+ * Defines a rectangle that can be displayed in the Canvas
+ */
+export const Rect = (props: IRectProps) => {
+    const { data, fill, stroke, stroke_weight } = props.shape;
+
+    // Fetch data — [x, y, width, height]
```
Comment (MathewMacDougall) [inline src/corner_kick/src/components/Canvas/shapes/Rect.tsx:18]: I believe the standard (for graphics stuff) is that `x, y` is the "top left" of the rectangle, right? Would still be good to mention that quickly.

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: src/corner_kick/src/components/LayersPanel/__stories__/index.tsx [hunk 3]
```diff
@@ -18,23 +18,23 @@ stories.add('with layers', () => (
         <LayersPanel
             layers={[
                 {
-                    id: 'friendly_robots',
                     name: 'Friendly Robots',
```
Comment (FSXAC) [inline src/corner_kick/src/components/LayersPanel/__stories__/index.tsx:21]: Are these the layer names from messages? 

If so, I currently have the following layer names defined on VM: "field", "ball", "ball_vel", "friendly_robot", "friendly_robot_vel", "enemy_robot", "enemy_robot_vel", "nav", "rule", "passing", "test", "misc". (util/visualizer_messenger/visualizer_messenger.cpp)

We either needs to have the same naming, or have some 1:1 mapping between layer name from the message, and layer name for UI.

Comment (FSXAC) [inline src/corner_kick/src/components/LayersPanel/__stories__/index.tsx:21]: Are these the layer names from messages? 

If so, I currently have the following layer names defined on VM: "field", "ball", "ball_vel", "friendly_robot", "friendly_robot_vel", "enemy_robot", "enemy_robot_vel", "nav", "rule", "passing", "test", "misc". (util/visualizer_messenger/visualizer_messenger.cpp)

We either needs to have the same naming, or have some 1:1 mapping between layer name from the message, and layer name for UI.

File: src/corner_kick/src/components/Canvas/shapes/Rect.tsx (hunk 2 above)
Comment (qgolsteyn) [reply src/corner_kick/src/components/Canvas/shapes/Rect.tsx:18]: Sure! I can add something quickly

File: src/corner_kick/src/components/LayersPanel/__stories__/index.tsx (hunk 3 above)
Comment (qgolsteyn) [reply src/corner_kick/src/components/LayersPanel/__stories__/index.tsx:21]: This is a fixture I made for my visual tests and doesn't make any assumptions regarding the layers of the AI. I can modify it however to be more in line with the layers in the AI

File: src/corner_kick/src/components/Canvas/Layer.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    // We select the type of shape to return based on the type property of the
+    // shape
+    const shapesElements = props.layer.shapes.map((shape, index) => {
+        switch (shape.type) {
+            case 'rect':
+                return <Rect key={`${props.layer.name}${index}`} shape={shape} />;
+            default:
+                return null;
```
Comment (garethellis0) [inline src/corner_kick/src/components/Canvas/Layer.tsx:32]: Could we log an error here? (No worries if the visualizer logging isn't quite there yet, just figured I'd ask).

File: src/corner_kick/src/components/LayersPanel/__stories__/index.tsx (hunk 3 above)
Comment (garethellis0) [reply src/corner_kick/src/components/LayersPanel/__stories__/index.tsx:21]: I don't think it's a huge issue here if this is just for the storybook. IIRC, the visualizer can handle whatever layers get passed to it right? It doesn't need to know anything about what layers the AI might send it?

File: src/corner_kick/src/types/visualizer.ts [hunk 5]
```diff
@@ -1,9 +1,58 @@
-/**
- * This file describes a layer datatype
+/*
+ * This file describes the datatypes used for the visualizer
```
Comment (garethellis0) [inline src/corner_kick/src/types/visualizer.ts:2]: Just a question, we only have 58 lines of types in the entire visualizer?

**Review by garethellis0 — APPROVED** [review]
This LGTM, just a couple questions.

File: src/corner_kick/src/components/LayersPanel/__stories__/index.tsx (hunk 3 above)
Comment (qgolsteyn) [reply src/corner_kick/src/components/LayersPanel/__stories__/index.tsx:21]: Yup this is correct

File: src/corner_kick/src/types/visualizer.ts (hunk 5 above)
Comment (qgolsteyn) [reply src/corner_kick/src/types/visualizer.ts:2]: No, this is just for the "visualizer" piece of the visualizer. We have more type files in the type directory

File: src/corner_kick/src/components/Canvas/Layer.tsx (hunk 4 above)
Comment (qgolsteyn) [reply src/corner_kick/src/components/Canvas/Layer.tsx:32]: Added #289 for discussion on this

File: src/corner_kick/src/components/LayersPanel/__stories__/index.tsx (hunk 3 above)
Comment (FSXAC) [reply src/corner_kick/src/components/LayersPanel/__stories__/index.tsx:21]: Gotcha! o7

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**